robot/power.py

```diff
@@ -53,13 +53,13 @@ def start_button_pressed(self):
         status = self.send_and_receive({})
         return status["start-button"]
 
-    def buzz(self, duration, note=None, frequency=None):
+    def buzz(self, duration, *, note=None, frequency=None):
         if note is None and frequency is None:
             raise ValueError("Either note or frequency must be provided")
+        if note is not None and frequency is not None:
+            raise ValueError("Only provide note or frequency")
         if note is not None:
-            if note not in self.BUZZ_NOTES:
-                raise KeyError("{} is an invalid note".format(note))
-            frequency = self.BUZZ_NOTES.get(note.lower())
+            frequency = self.BUZZ_NOTES[note.lower()]
         if frequency is None:
             raise ValueError("Invalid frequency")
```

Comment:
While we're tidying the arguments it might be nice to include the actual value of `frequency` in this error message.